src/main/java/org/broadinstitute/dsm/route/ClinicalKitsRoute.java

```diff
@@ -0,0 +1,91 @@
+package org.broadinstitute.dsm.route;
+
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.dto.kit.ClinicalKitDto;
+import org.broadinstitute.dsm.model.BSPKit;
+import org.broadinstitute.dsm.model.ParticipantWrapper;
+import org.broadinstitute.dsm.model.bsp.BSPKitInfo;
+import org.broadinstitute.dsm.model.bsp.BSPKitQueryResult;
+import org.broadinstitute.dsm.statics.RequestParameter;
+import org.broadinstitute.dsm.util.ElasticSearchUtil;
+import org.broadinstitute.dsm.util.NotificationUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class ClinicalKitsRoute implements Route {
+    private String FIRSTNAME = "firstName";
+    private String LASTNAME = "lastName";
+    private String DATE_OF_BIRtH = "dateOfBirth";
+
+    private static final Logger logger = LoggerFactory.getLogger(ClinicalKitsRoute.class);
+
+    private NotificationUtil notificationUtil;
+
+    public ClinicalKitsRoute(@NonNull NotificationUtil notificationUtil) {
+        this.notificationUtil = notificationUtil;
+    }
+
+    @Override
+    public Object handle(Request request, Response response) {
+        String kitLabel = request.params(RequestParameter.LABEL);
+        if (StringUtils.isBlank(kitLabel)) {
+            throw new RuntimeException("Please include a kit label as a path parameter");
+        }
+        BSPKit bspKit = new BSPKit();
+        if (!bspKit.canReceiveKit(kitLabel)) {
+            Optional<Object> result = Optional.ofNullable(bspKit.getKitStatus(kitLabel, notificationUtil));
+            if(!result.isPresent()){
+                response.status(404);
+            }
+            return result.orElse(response);
+        }
+
+        return getClinicalKit(kitLabel);
+    }
+
+    private ClinicalKitDto getClinicalKit(String kitLabel) {
+        logger.info("Checking label " + kitLabel);
+
+        // this method already sets the received time, check for exited and deactivation and special behaviour, and triggers DDP, we don't need a new
+        BSPKit bspKit = new BSPKit();
+        ClinicalKitDto clinicalKit = new ClinicalKitDto();
+        Optional<BSPKitInfo> maybeKitInfo = Optional.of(bspKit.receiveBSPKit(kitLabel, notificationUtil));
```

Comment:
the same here, return Optional from `receiveBSPKit` instead of wrapping it everywhere.